src/rust/inetstack/protocols/tcp/segment.rs

```diff
@@ -358,7 +362,8 @@ impl TcpHeader {
             num_options,
             option_list,
         };
-        buf.adjust(data_offset);
+        buf.adjust(data_offset)
+            .expect("buf should contain at least 'data_offset' bytes");
```

Comment:
Why can't we propagate the error?

Reply:
This exact error case is checked for and propagated to the caller earlier in the function.  So, this is a "shouldn't happen" case worthy of an assert (which this expect effectively is).

If we get too many of these situations we might want to consider adding adjust_unchecked() and trim_unchecked() calls to the DemiBuffer API to save a few cycles on the fast path.